### PR TITLE
WIP: Messaging: add URL previewer from Polari

### PR DESCRIPTION
**Pull request by andyholmes**
Add preview images for links in SMS messages.

See also: https://gitlab.gnome.org/GNOME/polari/-/merge_requests/134

### CODE REVIEW

**Comment by andyholmes** [comment]
cc @jtojnar 

There's still a few more things I'll add tonight before I merge, but this is pretty much it for the script execution. Line `#218` of `uri.js` is where the call is made.

File: src/utils/uri.js [hunk 1]
```diff
@@ -166,3 +167,72 @@ var SmsURI = class URI {
...
+
+        let nextData = this._urlQueue.shift();
+        if (nextData)
+            this._processData(nextData);
+    }
+
+    _generateThumbnail(data) {
+        let {filename, uri} = data;
+        this._subProc = Gio.Subprocess.new(
+            ['gjs', `${gsconnect.extdatadir}/utils/thumbnailer.js`, uri, filename],
```
Comment (jtojnar) [inline src/utils/uri.js]: Yeah, we prefer shebangs, as they are easier to patch automatically to a concrete path.

Comment (jtojnar) [reply src/utils/uri.js]: Though, why not just import that as a module? It should be possible to make it sync as well, should not it?

Comment (andyholmes) [reply src/utils/uri.js]: That's true, I've been playing with inlining it a bit. The catch is gjs still doesn't deal well with garbage collection complex objects like GtkWindow, so so far it has some pretty big memory increases if I run it in-process.

I'll work more at it and ping you again closer to merge-time.

**Comment by andyholmes** [comment]
Time to close this one down, it's not going anywhere

**Pull request closed without merging**